server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -1681,15 +1681,16 @@ def replace(
         step: "ReplaceStep",
     ) -> StepContext:
         col_field = prev_step_table[step.search_column]
-
         # Do a nested `replace` to replace many values on the same column
         replaced_col = col_field
+
+        case_ = Case()
         for old_name, new_name in step.to_replace:
-            replaced_col = functions.Replace(replaced_col, old_name, new_name)
+            case_ = case_.when(col_field == old_name, functions.Replace(replaced_col, old_name, new_name)).else_(
+                col_field
+            )
 
```

Comment:
```suggestion
            case_ = case_.when(col_field == old_name, LIteravalValue(new_name)).else_(col_field)
```

We don't want to replace text here

Reply:
I'm not sure about `LIteraval` 😄 
But true, I missed that!